2023/src/day5/almanac.rs

```diff
@@ -17,6 +17,30 @@ pub struct AlmanacEntry {
   pub kind: AmcEntryKind,
   pub amc_maps: Vec<AlmanacMap>,
 }
+impl AlmanacEntry {
+  pub fn get_dest(&self, src: u64) -> u64 {
+    for amc_map in self.amc_maps.iter() {
+      // check if the src is in the map range
+      let src_end = amc_map.src_start + amc_map.range_len;
+      if src >= amc_map.src_start 
+        && src < src_end
+      {
+        // calculate the offset if it exists
+        let offset: i64 = amc_map.dest_start as i64 - amc_map.src_start as i64;
+        let dest = src.clone() as i64 + offset;
```

Comment:
I don't believe the `.clone()` is necessary here, since all the primitive numeric types and `char` implement `Copy`  and that should happen here by default.  That shouldn't change anything about how the code is executed; it's mostly for whoever is reading it.  Whenever I see `.clone()`  a little warning light goes on in my head and I have to check what's being cloned and why.

Does the compiler complain without it?

Reply:
Nope - I had initially written passing `src` as a reference, and that ended up having a lifecycle error. This must've been left over from me trying to hack around that before just passing `u64` by value.

Removing the clone worked, nice catch. Thanks!